src/App.tsx

```diff
@@ -15,36 +17,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_BY_ALPH: number = 1;
+const SORT_BY_LENGTH: number = 2;
+
+const sortGoods = (goods: string[], key: number): string[] => {
+  switch (key) {
+    case SORT_BY_ALPH:
+      return goods.sort((a, b) => a.localeCompare(b));
+    case SORT_BY_LENGTH:
+      return goods.sort((a, b) => a.length - b.length);
+    default:
+      return goods;
+  }
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(0);
+  const [toReverse, setToReverse] = useState(false);
+
+  const goods = sortGoods(goodsFromServer.slice(), sortBy);
+
+  if (toReverse) {
+    goods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortBy !== SORT_BY_ALPH,
+          })}
+          onClick={() => setSortBy(SORT_BY_ALPH)}
```

Comment:
[UX] The 'Sort alphabetically' button doesn't provide any visual feedback to the user when it's active. Consider changing the style of the button when it's active to improve the user experience.